pkg/apis/serving/v1alpha1/cluster_local_model_types.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2023 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:openapi-gen=true
+type ClusterLocalModelSpec struct {
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="StorageUri is immutable"
+	// Original StorageUri
+	SourceModelUri string `json:"sourceModelUri" validate:"required"`
```

Comment:
maybe we need to consider supporting multiple source modelUris

Reply:
We can make it a list of models here or support multiple local model crs for one inference service.